lib/terraform/client.rb

```diff
@@ -50,8 +50,8 @@ def output(name:)
     end
 
     def version
-      execute command: factory.version_command do |value|
-        return ::Terraform::Version.create value: value
+      @version ||= execute command: factory.version_command do |value|
```

Comment:
One thing that I can think of is if it's cached across instances it would prevent testing multiple versions of terraform. Not sure if that would be possible otherwise or if that even makes sense to do or not.. but I could see it potentially being useful to someone.